File: java/src/main/java/dev/openfeature/flagd/evaluator/EvaluatorException.java
package dev.openfeature.flagd.evaluator;

/**
 * Exception thrown when flag evaluation fails.
 *
 * <p>This exception wraps underlying errors that may occur during:
 * <ul>
 *   <li>WASM module interaction
 *   <li>JSON serialization/deserialization
 *   <li>Memory allocation/deallocation
 *   <li>Flag configuration parsing
 * </ul>
 */
public class EvaluatorException extends Exception {

    /**
     * Creates a new evaluator exception with the specified message.
     *
     * @param message the error message
     */
    public EvaluatorException(String message) {
        super(message);
    }

    /**
     * Creates a new evaluator exception with the specified message and cause.
     *
     * @param message the error message
     * @param cause the underlying cause
     */
    public EvaluatorException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: dotnet/src/FlagdEvaluator/EvaluatorException.cs
namespace FlagdEvaluator;

/// <summary>
/// Exception thrown by the flag evaluator for WASM or evaluation errors.
/// </summary>
public class EvaluatorException : Exception
{
    public EvaluatorException(string message) : base(message) { }
    public EvaluatorException(string message, Exception innerException) : base(message, innerException) { }
}
